### DISCUSSION TITLE
How can I add my own syntax support to CotEditor?

### DISCUSSION THREAD

**Discussion by xycabcd** [coteditor/CotEditor#1639] [Q&A] [answered]
I see there's a CotEditor.app/Contents/Resources/Syntaxes with lot's of file for syntaxes of different languages in it.
Can I add my own? What's the syntax of this kind of file?
Also I see there's a SyntaxMap.json, what's this for?

**Comment by 1024jp** [comment, marked as answer]
You can add and edit your own syntaxes in the Format settings pane.

<img width="712" alt="Screenshot 2024-05-21 at 22 39 42" src="https://github.com/coteditor/CotEditor/assets/1165044/4bae8935-4b39-40a0-8249-59c8267dd7b4">

See the CotEditor User Guides for the details. You can launch it from the help button in the syntax editor.

<img width="792" alt="Screenshot 2024-05-21 at 22 41 02" src="https://github.com/coteditor/CotEditor/assets/1165044/97e2bff9-c419-4b8f-837f-a79603f4bcdd">

> Also I see there's a SyntaxMap.json, what's this for?

The file is used to speed up the detection of the proper syntax when opening a document.
It is just for bundled syntaxes. Custom user syntaxes don't require this file.

**Comment by xycabcd** [comment]
I already closed it because I see it.

**Reply by 1024jp** [reply]
I wrote this for users who may come here in the future with the same question.
